src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -773,6 +793,46 @@ private void handleReset(
             final long traceId = reset.trace();
             doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId);
         }
+
+        // TODO - SseEventFW
+        //                  .type = "challenge"
+        //                  .data = "{}"
+        // Received signalEx CHALLENGE from OAuth
+        private void handleSignal(
+            SignalFW signal)
+        {
+            final HttpSignalExFW httpSignalEx = httpSignalExRO.tryWrap(signal.buffer(),
+                                                                       signal.extension().offset(),
+                                                                       signal.extension().limit());
+            if (httpSignalEx != null && signal.extension().sizeof() == httpSignalExRO.limit())
+            {
+                final ListFW<HttpHeaderFW> httpHeaders = httpSignalEx.headers();
+                final Map<String, String> collectedHeaders = new HashMap<>();
+                httpHeaders.forEach(header -> collectedHeaders.put(header.name().asString(), header.value().asString()));
+
+                // TODO: need ":method" -> "method"
+                final JsonObject jsonHeaders = new JsonObject();
+                collectedHeaders.forEach(jsonHeaders::addProperty);
+                final String headersPayload = gson.toJson(jsonHeaders);
+
+                final JsonObject payload = new JsonObject();
+                payload.addProperty("headers", headersPayload);
+                final String jsonPayload = gson.toJson(payload);
```

Comment:
Replace lines above with `challenge` `JsonObject` which potentially has `method` and `headers` properties.